platforms/jvm/code-quality/src/main/java/org/gradle/api/plugins/quality/internal/CheckstyleInvoker.java

```diff
@@ -0,0 +1,279 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins.quality.internal;
+
+import com.google.common.collect.ImmutableMap;
+import groovy.namespace.QName;
+import groovy.util.Node;
+import groovy.util.NodeList;
+import groovy.xml.XmlParser;
+import org.apache.commons.io.IOUtils;
+import org.gradle.api.Action;
+import org.gradle.api.GradleException;
+import org.gradle.api.InvalidUserDataException;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.file.FileTree;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.internal.exceptions.MarkedVerificationException;
+import org.gradle.api.internal.project.antbuilder.AntBuilderDelegate;
+import org.gradle.api.plugins.quality.Checkstyle;
+import org.gradle.api.provider.Property;
+import org.gradle.internal.logging.ConsoleRenderer;
+import org.gradle.util.GradleVersion;
+import org.gradle.util.internal.GFileUtils;
+import org.gradle.util.internal.VersionNumber;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+import javax.annotation.Nullable;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static org.gradle.api.plugins.quality.internal.AntInvokeUtils.invoke;
+
+class CheckstyleInvoker implements Action<AntBuilderDelegate> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CheckstyleInvoker.class);
+
+    private static final String FAILURE_PROPERTY_NAME = "org.gradle.checkstyle.violations";
+    private static final String CONFIG_LOC_PROPERTY = "config_loc";
+
+    private final CheckstyleActionParameters parameters;
+
+    CheckstyleInvoker(CheckstyleActionParameters parameters) {
+        this.parameters = parameters;
+    }
+
+    @Override
+    @SuppressWarnings({"UnusedDeclaration", "CodeBlock2Expr"})
+    public void execute(AntBuilderDelegate ant) {
+        FileTree source = parameters.getSource().getAsFileTree();
+        boolean showViolations = parameters.getShowViolations().get();
+        int maxErrors = parameters.getMaxErrors().get();
+        int maxWarnings = parameters.getMaxWarnings().get();
+        Map<String, Object> configProperties = parameters.getConfigProperties().get();
+        boolean ignoreFailures = parameters.getIgnoreFailures().get();
+        RegularFile config = parameters.getConfig().get();
+        File configDir = parameters.getConfigDirectory().getAsFile().getOrNull();
```

Comment:
This should just be `get()`, it can't be `null` as we call `toString()` on it in the `checkstyle` configuration.